lib/idn.c

```diff
@@ -181,6 +181,18 @@ static CURLcode idn_decode(const char *input, char **output)
     result = CURLE_NOT_BUILT_IN;
 #elif defined(USE_WIN32_IDN)
   result = win32_idn_to_ascii(input, &decoded);
+#elif defined(HAVE_NETDB_H) && defined(__APPLE__)
+    /* let MacOS/iOS do this as part of DNS service */
+    struct hostent *p = gethostbyname(input);
```

Comment:
Perhaps check for `HAVE_GETHOSTBYNAME_R` or call `Curl_ipv4_resolve_r()`?

Reply:
Well, macOS and iOS have no gethostbyname_r as far as I know. 
Calling Curl_ipv4_resolve_r may be a possibility, so we use Curl_getaddrinfo_ex here, which uses getaddrinfo internally, skipping gethostbyname all together.